test/model/recent_senders_test.dart

```diff
@@ -142,6 +145,164 @@ void main() {
     });
   });
 
+  group('RecentSenders.handleUpdateMessageEvent', () {
+    late PerAccountStore store;
+    late RecentSenders model;
+
+    final origChannel = eg.stream(); final newChannel = eg.stream();
+    final origTopic   = 'origTopic'; final newTopic   = 'newTopic';
+    final userX       = eg.user();   final userY      = eg.user();
+
+    Future<void> prepare(List<Message> messages) async {
+      store = eg.store();
+      await store.addMessages(messages);
+      await store.addStreams([origChannel, newChannel]);
+      await store.addUsers([userX, userY]);
+      model = store.recentSenders;
+    }
+
+    List<StreamMessage> copyMessagesWith(Iterable<StreamMessage> messages, {
+      ZulipStream? newChannel,
+      String? newTopic,
+    }) {
+      assert(newChannel != null || newTopic != null);
+      return messages.map((message) => StreamMessage.fromJson(
+        message.toJson()
+          ..['stream_id'] = newChannel?.streamId ?? message.streamId
+          // See [StreamMessage.displayRecipient] for why this is needed.
+          ..['display_recipient'] = newChannel?.name ?? message.displayRecipient!
+
+          ..['subject'] = newTopic ?? message.topic
+      )).toList();
+    }
+
+    test('move a conversation entirely, with additional unknown messages', () async {
+      final messages = List.generate(10, (i) => eg.streamMessage(
+        stream: origChannel, topic: origTopic, sender: userX));
+      await prepare(messages);
+      final unknownMessages = List.generate(10, (i) => eg.streamMessage(
+        stream: origChannel, topic: origTopic, sender: userX));
+      checkMatchesMessages(model, messages);
+
+      final messageIdsByUserInStreamBefore = model.streamSenders
+        [origChannel.streamId]![userX.userId]!.ids;
+      final messageIdsByUserInTopicBefore = model.topicSenders
+        [origChannel.streamId]![TopicName(origTopic)]![userX.userId]!.ids;
```

Comment:
These line breaks feel a little awkward; can we try something like:

```suggestion
      final messageIdsByUserInStreamBefore =
        model.streamSenders[origChannel.streamId]![userX.userId]!.ids;
      final messageIdsByUserInTopicBefore =
        model.topicSenders[origChannel.streamId]![eg.t(origTopic)]![userX.userId]!.ids;
```

(here and elsewhere)

Reply:
Didn't know that `eg.t` exists! Thanks, that's helpful (it still went over by 7 characters `!.ids);`, but it's a bit better than 12 with `TopicName`).